examples/texture_cubemap/source/texture_cubemap.cpp

```diff
@@ -0,0 +1,449 @@
+#include <gvk.hpp>
+#include <imgui.h>
+#include "image_resource.hpp"
+
+#define USE_SERIALIZER
+
+class texture_cubemap_app : public gvk::invokee
+{
+	struct data_for_draw_call
+	{
+		std::vector<glm::vec3> mPositions;
+		//std::vector<glm::vec2> mTexCoords;
+		std::vector<glm::vec3> mNormals;
+		std::vector<uint32_t> mIndices;
+
+		avk::buffer mPositionsBuffer;
+		//avk::buffer mTexCoordsBuffer;
+		avk::buffer mNormalsBuffer;
+		avk::buffer mIndexBuffer;
+
+		//int mMaterialIndex;
+	};
+
+	struct view_projection_matrices {
+		glm::mat4 mProjectionMatrix;
+		glm::mat4 mModelViewMatrix;
+		glm::mat4 mInverseModelViewMatrix;
+		float mLODBias = 0.0f;
+	};
+
+public: // v== avk::invokee overrides which will be invoked by the framework ==v
+	texture_cubemap_app(avk::queue& aQueue)
+		: mQueue{ &aQueue }
+		, mScale{1.0f, 1.0f, 1.0f}
+	{}
+
+	void initialize() override
+	{
+		mInitTime = std::chrono::high_resolution_clock::now();
+
+		// Create a descriptor cache that helps us to conveniently create descriptor sets:
+		mDescriptorCacheSkybox = gvk::context().create_descriptor_cache();
+		mDescriptorCacheReflect = gvk::context().create_descriptor_cache();
+
+		// Test caching with cubemap
+		const std::string cacheFilePath("assets/cubemap.cache");
+		auto cacheFileExists = gvk::does_cache_file_exist(cacheFilePath);
+		auto serializer = cacheFileExists ?
+			gvk::serializer(gvk::serializer::deserialize(cacheFilePath)) :
+			gvk::serializer(gvk::serializer::serialize(cacheFilePath));
+
+		// Load a cubemap image file
+		// The cubemap texture coordinates start in the upper right corner of the skybox faces,
+		// which coincides with the memory layout of the textures. Therefore we don't need to flip them along the y axis.
+		// Note that lookup operations in a cubemap are defined in a left-handed coordinate system,
+		// i.e. when looking at the positive Z face from inside the cube, the positive X face is to the right.
+
+		gvk::image_resource cubemap_image_resource;
+
+		// Load the textures for all cubemap faces from one file (.ktx or .dds format), or from six individual files
+		bool load_single_file = true;
+		if (load_single_file) {
+			bool load_dds = false;
+			if (load_dds)
+			{
+				cubemap_image_resource = gvk::image_resource(new gvk::image_resource_t("assets/yokohama_at_night-All-Mipmaps-Srgb-RGBA8-DXT1-SRGB.dds", true, true, false));
```

Comment:
The classes `image_resource` and and `image_resource_t` are intended to follow the pattern of other resource types, e.g. `buffer` and `buffer_t`. There was a problem compiling `image_resource` as an `owning_resource<T>`, but I don't remember exactly what it was. I will try again to see what the issue was. The heap allocation shouldn't actually be necessary.
In general, it is also possible to simply use `create_cubemap_from_file(filename, ...)`, which will create the intermediate `image_resource` itself. This is also the more common use case (the example should probably use this as well, creating the object explicitly is more a remnant of me testing the implementation).
I have looked into both the Config and Meta pattern, but am not sure if either apply. Having a single constructor and `create_cubemap_from_file()` would be nice, but it must support loading the cubemap from a single file with six faces and from six files with a face each. 
I think a Config pattern with a variable number of filenames as input would be suitable. Meta doesn't really fit here because all filenames are the same type, and they don't contain any meta-information about the resulting cubemap (which is exactly the same, regardless of whether it was loaded from one or more files).

Reply:
I have now defined image_resource as an avk::owning_resource<image_resource_t> and adapted the code.

Maybe the meta or config patterns could be used to define different aspects of create_{image,cubemap}_from_* and thus reduce the large number of similar function calls. This way it should also be possible to support creating image arrays with a nice interface. I would have to work out the details first but it would be an interesting approach.

Reply:
Hmm.. I think, I have addressed all these points in my other comment (above, in `image_resource.hpp`) and explained why I think that the naming is not optimal because in _Auto-Vk_, a "resource" usually refers to a "Vulkan GPU resource", but this `image_resource_t` here is actually a "CPU resource" which only needs to live temporary - for as long as the "Vulkan GPU resource" is fully created.

I haven't looked at the updated code yet. If it is ready for re-review, please request another review! (Should be possible on GitHub somewhere to request that.)